### PR TITLE
Enable linting for dropdown.vue

### PR DESCRIPTION
**Pull request by jonahtanjz**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [ ] Bug fix
- [ ] Feature addition or enhancement
- [x] Code maintenance
- [ ] Others, please explain:

**Overview of changes:**
This is a follow-up PR to #1455. Linting for `dropdown.vue` is enabled and code is refactored to be consistent with the lint rules. 

**Anything you'd like to highlight / discuss:**
Previously, there is a `class` which is set to `null` in the props. Not sure what is the function of it as it seems to work fine after removal. If it is needed, will proceed to add it back. 

**Testing instructions:**
`npm run test`

**Proposed commit message: (wrap lines at 72 characters)**
Enable linting for dropdown.vue

Currently, stylelint is only enabled for css files. Vue components may
also contain <style> tags with valid css but these are not linted. 

Let's enable stylelint css for vue files to provide linting to them as well.

---

**Checklist:** :ballot_box_with_check:

- [x] Updated the documentation for feature additions and enhancements
- [x] Added tests for bug fixes or features
- [x] Linked all related issues
- [x] No blatantly unrelated changes
- [x] Pinged someone for a review!

### CODE REVIEW

**Comment by jonahtanjz** [comment]
@ang-zeyu This PR is ready for review. Would appreciate if you can look through and give some feedbacks 😃

File: packages/vue-components/src/Dropdown.vue [hunk 1]
```diff
@@ -120,58 +132,58 @@ export default {
...
-  position: absolute;
-  clip: rect(0 0 0 0);
-  overflow: hidden;
-  margin: -1px;
-  height: 1px;
-  width: 1px;
-  padding: 0;
-  border: 0;
-}
+    .secret {
```
Comment (ang-zeyu) [inline packages/vue-components/src/Dropdown.vue:168]: looks good!

just one nit: let's enable and fix stylelint separately (for all `.vue` files that are es-linted)

Comment (jonahtanjz) [reply packages/vue-components/src/Dropdown.vue:168]: Sure 👍 
Should I do it in this PR or open a separate one?

Comment (ang-zeyu) [reply packages/vue-components/src/Dropdown.vue:168]: let's do it separately, iirc none of the .vue files are validated in our ci tests - might be quite a few changes

Comment (jonahtanjz) [reply packages/vue-components/src/Dropdown.vue:168]: Alright. I've enabled stylelint for all `.vue` files but added all of them to `.stylelintignore` except for `dropdown.vue` for now. Will do it separately for all the other files in the next PR :)

File: .stylelintignore [hunk 2]
```diff
@@ -2,3 +2,28 @@
 **/_site/**/*.css
 packages/cli/test/functional/**/expected/**/*.css
 packages/vue-components/**/*.css
+
+# --- packages/vue-components .vue files ---
+
+packages/vue-components/src/Navbar.vue
```
Comment (ang-zeyu) [inline .stylelintignore:8]: meant to shift all of stylelint to another PR, so you wouldn't have to write this huge ignore list here 😅

but am fine with this as well.

**Review by ang-zeyu — APPROVED** [review]
Thanks! @jonahtanjz

File: .stylelintignore (hunk 2 above)
Comment (ang-zeyu) [reply .stylelintignore:8]: > Enable linting for dropdown.vue

let's update the message body to highlight vue css lint enabling as well

Comment (jonahtanjz) [reply .stylelintignore:8]: Thanks @ang-zeyu. Updated the message body.

**Review by ang-zeyu — APPROVED** [review]

**Pull request merged**